pkg/resource/deploy/step_generator.go

```diff
@@ -376,6 +376,13 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, err
 			return []Step{NewUpdateStep(sg.plan, event, old, new, diff.StableKeys)}, nil
 		}
 
+		// If resource was unchanged, but there were initialization errors, generate an empty update
+		// step to attempt to "continue" awaiting initialization.
+		if len(old.InitErrors) > 0 {
```

Comment:
Is the idea here that by awaiting resources that failed to initialize we'll end up either in an error state (failing the deployment) or a success state?

Reply:
yes.